core/templates/components/smooth-height/smooth-height-animation.component.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the smooth height change animation.
+ */
+
+import { ElementRef, HostBinding, Component, Input, OnChanges } from '@angular/core';
+import { animate, style, transition, trigger } from '@angular/animations';
+
+@Component({
+  selector: 'oppia-smooth-height-animation',
```

Comment:
Sorry -- could you go with `-animator` instead? On further thought "animation" doesn't really fit because it is not a thing, it refers to the technique of animating.

At least with `-animator` the idea is that the directive is doing or responsible for the animating.

Reply:
Done